eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -420,6 +431,6 @@ private[channel] object ChannelCodecs3 {
     .typecase(0x03, Codecs.DATA_SHUTDOWN_COMPAT_03_Codec)
     .typecase(0x02, Codecs.DATA_NORMAL_COMPAT_02_Codec)
     .typecase(0x01, Codecs.DATA_WAIT_FOR_CHANNEL_READY_COMPAT_01_Codec)
-    .typecase(0x00, Codecs.DATA_WAIT_FOR_FUNDING_CONFIRMED_Codec)
+    .typecase(0x00, Codecs.DATA_WAIT_FOR_FUNDING_CONFIRMED_COMPAT_00_Codec)
```

Comment:
This is really footgunny and I got bitten before: all `Codecs.DATA_xxx_Codec` should be numbered , even the current ones, so the codec names are immutable.

Reply:
Fixed by bfba9e41193da5fbe76a7685485e131293f32789.